tools/crm_verify.c

```diff
@@ -162,125 +163,58 @@ main(int argc, char **argv)
-    if (status == NULL) {
-        create_xml_node(cib_object, XML_CIB_TAG_STATUS);
-    }
-
-    if (pcmk__validate_xml(cib_object, NULL, (xmlRelaxNGValidityErrorFunc) out->err, out) == FALSE) {
-        pcmk__config_err("CIB did not pass schema validation");
-        free_xml(cib_object);
-        cib_object = NULL;
-
-    } else if (cli_config_update(&cib_object, NULL, FALSE) == FALSE) {
-        crm_config_error = TRUE;
-        free_xml(cib_object);
-        cib_object = NULL;
-        out->err(out, "The cluster will NOT be able to use this configuration.\n"
-                 "Please manually update the configuration to conform to the %s syntax.",
-                 xml_latest_schema());
-    }
-
     scheduler = pe_new_working_set();
+
     if (scheduler == NULL) {
         rc = errno;
         g_set_error(&error, PCMK__RC_ERROR, rc,
                     "Could not allocate scheduler data: %s", pcmk_rc_str(rc));
         goto done;
     }
+    
     scheduler->priv = out;
 
-    /* Process the configuration to set crm_config_error/crm_config_warning.
-     *
-     * @TODO Some parts of the configuration are unpacked only when needed (for
-     * example, action configuration), so we aren't necessarily checking those.
-     */
-    if (cib_object != NULL) {
-        unsigned long long flags = pcmk_sched_no_counts|pcmk_sched_no_compat;
-
-        if ((status == NULL) && !options.use_live_cib) {
-            // No status available, so do minimal checks
-            flags |= pcmk_sched_validate_only;
-        }
-        pcmk__schedule_actions(cib_object, flags, scheduler);
-    }
-    pe_free_working_set(scheduler);
+    rc = pcmk__verify(scheduler, out, cib_object);
 
-    if (crm_config_error) {
-        rc = pcmk_rc_schema_validation;
-
-        if (args->verbosity > 0 || pcmk__str_eq(args->output_ty, "xml", pcmk__str_none)) {
-            g_set_error(&error, PCMK__RC_ERROR, rc,
-                        "Errors found during check: config not valid");
-        } else {
-            g_set_error(&error, PCMK__RC_ERROR, rc,
-                        "Errors found during check: config not valid\n-V may provide more details");
-        } 
-
-    } else if (crm_config_warning) {
-        rc = pcmk_rc_schema_validation;
-
-        if (args->verbosity > 0 || pcmk__str_eq(args->output_ty, "xml", pcmk__str_none)) {
-            g_set_error(&error, PCMK__RC_ERROR, rc,
-                        "Warnings found during check: config may not be valid");
-        } else {
-            g_set_error(&error, PCMK__RC_ERROR, rc,
-                        "Warnings found during check: config may not be valid\n-V may provide more details");
-        }
+    if (rc == pcmk_rc_schema_validation) {
+      if (args->verbosity > 0) {
+          out->err(out, "Errors found during check: config not valid");
+      } else {
+          out->err(out, "Errors found during check: config not valid\n-V may provide more details");
+      } 
```

Comment:
You'll need to check crm_config_err/warn to know whether to say Errors or Warnings here (you can use a variable and %s)